serde/src/private/de.rs

```diff
@@ -298,7 +302,7 @@ mod content {
             // Untagged and internally tagged enums are only supported in
             // self-describing formats.
             let visitor = ContentVisitor { value: PhantomData };
-            deserializer.__deserialize_content(actually_private::T, visitor)
+            deserializer.deserialize_any(visitor)
```

Comment:
Another way to deal with that is to remove `actually_private::T` from the method signature which is needed only to prevent calling this method from the other crates. The method with the similar intention, `Visitor::__private_visit_untagged_option` does not have such seal and survived separation without problems.

Reply:
This sounds like a change we could do on master and consider in isolation from this PR